doc/RFCs/APInotification.md

```diff
@@ -0,0 +1,245 @@
+# Caliopen #
+
+## Notifications
+
+### Objectif de ce document :
+
+Décrire le format des messages échangés entre le frontend et le backend
+
+Décrire le protocole utilisé pour l'échange de ces messages
+
+Lister les fonctionnalités de notification exposées par le backend pour le frontend (API)
+
+Décrire le mode de gestion de la file d'attente des notifications côté backend
+
+### Format des messages de notification :
+###### format de la réponse retournée par le backend sur /api/v2/notifications :
+
+```yaml  
+notifications:  
+  - from: string         // backend entity that's emitting the message. (contacts facility, email facility, etc.)  
+    to: string           // frontend service/component to which message is sent. (api-location, api-patch, contact, event-manager, etc.)  
+    id: string           // universally unique id to unambiguously identify a notification message.  
+    type: string         // a single word to describe message's type and give indication of importance level (event, info, feedback, warning, teaser, error, alert, etc.)  
+    reference: string    // (optional) a reference number previously sent by frontend to link current notification to a previous action/event.  
+    timestamp: int       // unix timestamp at which backend created the notification  
+    body: string         // could be a simple word or a more complex structure like a json, depending of the notification.  
+```
+
+##### Les messages de notification comportent des 'headers' et un 'body'.
+
+les 'headers' (`from`, `to`, `id`, `type`, `references`) permettent d'identifier la notification et de la classifier.
+
+le `body` comporte le message/payload de la notification
+
+les headers `from` et `to` servent à identifier les composants du backend (from) et du frontend (to) concernés par la notification.
+
+le header `type` permet d'identifier immédiatement le type de notification : event, info, feedback, warning, teaser, error, alert, etc.
+
+le header `reference` pourrait servir à lier une notification à un évènement ou une action initiée par le front afin que celui identifie le feedback du backend sur le déroulement de cette action.
+
+le header `timestamp` est le timestamp unix du moment où la notification a été créée par le backend. Il pourrait permettre au frontend d'indiquer une fourchette horaire lors du `GET /v2/nofications?from=00000000&to=000000000`
+
+
+
+#### Exemples de notifications :
+
+##### notification de réception de email(s) :
+
+```yaml  
+notifications:  
+  - from: email-facility  
```

Comment:
everything is a facility, so I don't think it's needed to have this word in description